diff -Naur gst-plugins-bad-1.14.4.old/configure.ac gst-plugins-bad-1.14.4/configure.ac
--- gst-plugins-bad-1.14.4.old/configure.ac	2018-10-03 00:04:23.000000000 +0200
+++ gst-plugins-bad-1.14.4/configure.ac	2019-01-05 21:20:55.454975446 +0100
@@ -945,11 +945,15 @@
       CUDA_LIBS="-L$CUDA_PREFIX/lib -L$CUDA_PREFIX/lib64 -L$CUDA_PREFIX/lib/stubs -L$CUDA_PREFIX/lib64/stubs -lcuda -lcudart"
     fi
   else
-    PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
-      PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
-        PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
-          PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
-            AC_MSG_WARN([Could not find cuda headers/libraries])])])])])
+    PKG_CHECK_MODULES([CUDA], [cuda-10.0 cudart-10.0],, [
+      PKG_CHECK_MODULES([CUDA], [cuda-9.2 cudart-9.2],, [
+        PKG_CHECK_MODULES([CUDA], [cuda-9.1 cudart-9.1],, [
+          PKG_CHECK_MODULES([CUDA], [cuda-9.0 cudart-9.0],, [
+            PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
+              PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
+                PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
+                  PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
+                    AC_MSG_WARN([Could not find cuda headers/libraries])])])])])])])])])
   fi
 
   HAVE_CUDA_H=no
@@ -977,11 +981,8 @@
 dnl *** NVDEC ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_NVDEC, true)
 AG_GST_CHECK_FEATURE(NVDEC, [nvdec], nvdec, [
-  HAVE_NVCUVID_H=no
   save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CUDA_CFLAGS $save_CPPFLAGS"
-  AC_CHECK_HEADER([nvcuvid.h], [HAVE_NVCUVID_H=yes],
-      AC_MSG_WARN([Could not find nvcuvid.h]))
   CPPFLAGS=$save_CPPFLAGS
 
   HAVE_NVCUVID=no
@@ -991,7 +992,7 @@
       AC_MSG_WARN([Could not find library nvcuvid]))
   LIBS="$save_LIBS"
 
-  if test "x$HAVE_NVCUVID_H" = "xyes" -a "x$HAVE_NVCUVID" = "xyes"; then
+  if test "x$HAVE_NVCUVID" = "xyes"; then
     HAVE_NVDEC=yes
   else
     HAVE_NVDEC=no
diff -Naur gst-plugins-bad-1.14.4.old/sys/nvdec/gstnvdec.c gst-plugins-bad-1.14.4/sys/nvdec/gstnvdec.c
--- gst-plugins-bad-1.14.4.old/sys/nvdec/gstnvdec.c	2018-05-15 13:08:18.000000000 +0200
+++ gst-plugins-bad-1.14.4/sys/nvdec/gstnvdec.c	2019-01-05 21:20:55.478975866 +0100
@@ -31,9 +31,6 @@
 
 #include "gstnvdec.h"
 
-#include <gst/gl/gstglfuncs.h>
-#include <cudaGL.h>
-
 typedef enum
 {
   GST_NVDEC_QUEUE_ITEM_TYPE_SEQUENCE,
diff -Naur gst-plugins-bad-1.14.4.old/sys/nvdec/gstnvdec.h gst-plugins-bad-1.14.4/sys/nvdec/gstnvdec.h
--- gst-plugins-bad-1.14.4.old/sys/nvdec/gstnvdec.h	2018-03-23 21:45:15.000000000 +0100
+++ gst-plugins-bad-1.14.4/sys/nvdec/gstnvdec.h	2019-01-05 21:28:40.578117652 +0100
@@ -29,7 +29,9 @@
 #define __GST_NVDEC_H__
 
 #include <gst/gl/gl.h>
+#include <gst/gl/gstglfuncs.h>
 #include <nvcuvid.h>
+#include <cudaGL.h>
 
 G_BEGIN_DECLS
 
diff -Naur gst-plugins-bad-1.14.4.old/sys/nvdec/Makefile.am gst-plugins-bad-1.14.4/sys/nvdec/Makefile.am
--- gst-plugins-bad-1.14.4.old/sys/nvdec/Makefile.am	2018-03-23 21:45:15.000000000 +0100
+++ gst-plugins-bad-1.14.4/sys/nvdec/Makefile.am	2019-01-05 21:20:55.477975849 +0100
@@ -19,7 +19,8 @@
 	$(GST_GL_LIBS) \
 	$(GST_PBUTILS_LIBS) \
 	$(GST_VIDEO_LIBS) \
-	$(GST_LIBS) \
-	$(CUDA_LIBS) -lnvcuvid
+	$(GST_LIBS)
+
+libgstnvdec_la_LIBADD += $(CUDA_LIBS) -lnvcuvid
 
 libgstnvdec_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
