diff -Naur gst-plugins-bad-1.14.1.old/configure.ac gst-plugins-bad-1.14.1/configure.ac
--- gst-plugins-bad-1.14.1.old/configure.ac	2019-03-03 14:22:42.855535770 +0100
+++ gst-plugins-bad-1.14.1/configure.ac	2019-03-03 14:22:57.876805180 +0100
@@ -1422,12 +1422,14 @@
 dnl *** fdkaac ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_FDK_AAC, true)
 AG_GST_CHECK_FEATURE(FDK_AAC, [fdkaac plugin], fdkaac, [
-  PKG_CHECK_MODULES(FDK_AAC, fdk-aac >= 0.1.4, [
-    HAVE_FDK_AAC="yes",
-    HAVE_FDK_AAC_0_1_4="yes"
+  PKG_CHECK_MODULES(FDK_AAC, fdk-aac, HAVE_FDK_AAC="yes", HAVE_FDK_AAC="no")
+
+  PKG_CHECK_EXISTS(fdk-aac >= 2.0.0, [
+    AC_DEFINE(HAVE_FDK_AAC_2_0_0, 1, [Define if we have fdk-aac >= 2.0.0])
+  ])
+
+  PKG_CHECK_EXISTS(fdk-aac >= 0.1.4, [
     AC_DEFINE(HAVE_FDK_AAC_0_1_4, 1, [Define if we have fdk-aac >= 0.1.4])
-  ],[
-    PKG_CHECK_MODULES(FDK_AAC, fdk-aac, HAVE_FDK_AAC="yes", HAVE_FDK_AAC="no")
   ])
 ])
 
diff -Naur gst-plugins-bad-1.14.1.old/ext/fdkaac/gstfdkaacdec.c gst-plugins-bad-1.14.1/ext/fdkaac/gstfdkaacdec.c
--- gst-plugins-bad-1.14.1.old/ext/fdkaac/gstfdkaacdec.c	2019-03-03 14:22:42.852535716 +0100
+++ gst-plugins-bad-1.14.1/ext/fdkaac/gstfdkaacdec.c	2019-03-03 16:54:38.320267698 +0100
@@ -151,17 +151,13 @@
     gst_buffer_unref (codec_data);
   }
 
+  /* Choose WAV channel mapping to get interleaving even with libfdk-aac 2.0.0
+   * The pChannelIndices retain the indices from the standard MPEG mapping so
+   * we're agnostic to the actual order. */
   if ((err =
           aacDecoder_SetParam (self->dec, AAC_PCM_OUTPUT_CHANNEL_MAPPING,
-              0)) != AAC_DEC_OK) {
-    GST_ERROR_OBJECT (self, "Failed to set output channel mapping: %d", err);
-    return FALSE;
-  }
-
-  if ((err =
-          aacDecoder_SetParam (self->dec, AAC_PCM_OUTPUT_INTERLEAVED,
               1)) != AAC_DEC_OK) {
-    GST_ERROR_OBJECT (self, "Failed to set interleaved output: %d", err);
+    GST_ERROR_OBJECT (self, "Failed to set output channel mapping: %d", err);
     return FALSE;
   }
 
diff -Naur gst-plugins-bad-1.14.1.old/ext/fdkaac/gstfdkaacenc.c gst-plugins-bad-1.14.1/ext/fdkaac/gstfdkaacenc.c
--- gst-plugins-bad-1.14.1.old/ext/fdkaac/gstfdkaacenc.c	2019-03-03 14:22:42.852535716 +0100
+++ gst-plugins-bad-1.14.1/ext/fdkaac/gstfdkaacenc.c	2019-03-03 14:23:21.561229962 +0100
@@ -71,10 +71,6 @@
   GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER,
           GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
           GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT}}, {
-    3, MODE_2_1, {
-  GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
-          GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT,
-          GST_AUDIO_CHANNEL_POSITION_LFE1}}, {
     4, MODE_1_2_1, {
   GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER,
           GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
@@ -93,6 +89,41 @@
           GST_AUDIO_CHANNEL_POSITION_SIDE_LEFT,
           GST_AUDIO_CHANNEL_POSITION_SIDE_RIGHT,
           GST_AUDIO_CHANNEL_POSITION_LFE1}}
+#ifdef HAVE_FDK_AAC_2_0_0
+  , {
+    /* MPEG 11: 6.1 Surround */
+    7, MODE_6_1, {
+      GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT,
+          /* FIXME: SURROUND instead of REAR is more to spec */
+  GST_AUDIO_CHANNEL_POSITION_REAR_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_REAR_RIGHT,
+          GST_AUDIO_CHANNEL_POSITION_REAR_CENTER,
+          GST_AUDIO_CHANNEL_POSITION_LFE1}}, {
+    /* MPEG 12: 7.1 Surround */
+    8, MODE_7_1_BACK, {
+      GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT,
+          /* FIXME: SURROUND instead of SIDE is more to spec */
+  GST_AUDIO_CHANNEL_POSITION_SIDE_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_SIDE_RIGHT,
+          GST_AUDIO_CHANNEL_POSITION_REAR_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_REAR_RIGHT,
+          GST_AUDIO_CHANNEL_POSITION_LFE1}}, {
+    /* MPEG 14: 5.1.2 Surround */
+    8, MODE_7_1_TOP_FRONT, {
+      GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT,
+          /* FIXME: SURROUND instead of REAR is more to spec */
+  GST_AUDIO_CHANNEL_POSITION_REAR_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_REAR_RIGHT,
+          GST_AUDIO_CHANNEL_POSITION_LFE1,
+          GST_AUDIO_CHANNEL_POSITION_TOP_FRONT_LEFT,
+          GST_AUDIO_CHANNEL_POSITION_TOP_FRONT_RIGHT}}
+#endif
 #ifdef HAVE_FDK_AAC_0_1_4
   , {
     8, MODE_7_1_REAR_SURROUND, {
diff -Naur gst-plugins-bad-1.14.1.old/ext/fdkaac/meson.build gst-plugins-bad-1.14.1/ext/fdkaac/meson.build
--- gst-plugins-bad-1.14.1.old/ext/fdkaac/meson.build	2019-03-03 14:22:42.852535716 +0100
+++ gst-plugins-bad-1.14.1/ext/fdkaac/meson.build	2019-03-03 14:22:57.878805215 +0100
@@ -8,6 +8,11 @@
     fdkaac_defines += ['-DHAVE_FDK_AAC_0_1_4']
   endif
 
+  # API v2
+  if fdkaac_dep.version().version_compare('>=2.0.0')
+    fdkaac_defines += ['-DHAVE_FDK_AAC_2_0_0']
+  endif
+
   gstfdkaac = library('gstfdkaac',
     ['plugin.c', 'gstfdkaacenc.c', 'gstfdkaacdec.c'],
     c_args : gst_plugins_bad_args + fdkaac_defines,
