diff -Naur gst-plugins-bad-1.16.2.old/configure.ac gst-plugins-bad-1.16.2/configure.ac
--- gst-plugins-bad-1.16.2.old/configure.ac	2020-11-18 17:21:49.226653549 +0100
+++ gst-plugins-bad-1.16.2/configure.ac	2020-11-18 17:22:09.851967174 +0100
@@ -1784,7 +1784,7 @@
 dnl *** neon ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_NEON, true)
 AG_GST_CHECK_FEATURE(NEON, [neon http client plugins], neonhttpsrc, [
-  PKG_CHECK_MODULES(NEON, neon >= 0.27.0  neon <= 0.30.99, HAVE_NEON="yes", [
+  PKG_CHECK_MODULES(NEON, neon >= 0.27.0  neon <= 0.31.99, HAVE_NEON="yes", [
     HAVE_NEON="no"
   ])
   AC_SUBST(NEON_CFLAGS)
diff -Naur gst-plugins-bad-1.16.2.old/ext/neon/meson.build gst-plugins-bad-1.16.2/ext/neon/meson.build
--- gst-plugins-bad-1.16.2.old/ext/neon/meson.build	2020-11-18 17:21:49.245653838 +0100
+++ gst-plugins-bad-1.16.2/ext/neon/meson.build	2020-11-18 17:21:56.437763200 +0100
@@ -1,4 +1,4 @@
-neon_dep = dependency('neon', version: ['>= 0.27', '<= 0.30.99'],
+neon_dep = dependency('neon', version: ['>= 0.27', '<= 0.31.99'],
                       required : get_option('neon'))
 
 if neon_dep.found()
