diff -Naur gst-plugins-bad-1.16.2.old/configure.ac gst-plugins-bad-1.16.2/configure.ac
--- gst-plugins-bad-1.16.2.old/configure.ac	2020-11-18 16:43:19.031261418 +0100
+++ gst-plugins-bad-1.16.2/configure.ac	2020-11-18 17:01:32.228469291 +0100
@@ -938,16 +938,18 @@
       CUDA_LIBS="-L$CUDA_PREFIX/lib -L$CUDA_PREFIX/lib64 -L$CUDA_PREFIX/lib/stubs -L$CUDA_PREFIX/lib64/stubs -lcuda -lcudart"
     fi
   else
-    PKG_CHECK_MODULES([CUDA], [cuda-10.1 cudart-10.1],, [
-      PKG_CHECK_MODULES([CUDA], [cuda-10.0 cudart-10.0],, [
-        PKG_CHECK_MODULES([CUDA], [cuda-9.2 cudart-9.2],, [
-          PKG_CHECK_MODULES([CUDA], [cuda-9.1 cudart-9.1],, [
-            PKG_CHECK_MODULES([CUDA], [cuda-9.0 cudart-9.0],, [
-              PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
-                PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
-                  PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
-                    PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
-                      AC_MSG_WARN([Could not find cuda headers/libraries])])])])])])])])])])
+    PKG_CHECK_MODULES([CUDA], [cuda-11.1 cudart-11.1],, [
+      PKG_CHECK_MODULES([CUDA], [cuda-11.0 cudart-11.0],, [
+        PKG_CHECK_MODULES([CUDA], [cuda-10.1 cudart-10.1],, [
+          PKG_CHECK_MODULES([CUDA], [cuda-10.0 cudart-10.0],, [
+            PKG_CHECK_MODULES([CUDA], [cuda-9.2 cudart-9.2],, [
+              PKG_CHECK_MODULES([CUDA], [cuda-9.1 cudart-9.1],, [
+                PKG_CHECK_MODULES([CUDA], [cuda-9.0 cudart-9.0],, [
+                  PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
+                    PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
+                      PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
+                        PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
+                          AC_MSG_WARN([Could not find cuda headers/libraries])])])])])])])])])])])])
   fi
 
   HAVE_CUDA_H=no
