diff -Naur gst-plugins-bad-1.14.5.old/configure.ac gst-plugins-bad-1.14.5/configure.ac
--- gst-plugins-bad-1.14.5.old/configure.ac	2020-11-30 13:42:27.555443299 +0100
+++ gst-plugins-bad-1.14.5/configure.ac	2020-12-08 17:15:33.027463794 +0100
@@ -945,11 +945,18 @@
       CUDA_LIBS="-L$CUDA_PREFIX/lib -L$CUDA_PREFIX/lib64 -L$CUDA_PREFIX/lib/stubs -L$CUDA_PREFIX/lib64/stubs -lcuda -lcudart"
     fi
   else
-    PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
-      PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
-        PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
-          PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
-            AC_MSG_WARN([Could not find cuda headers/libraries])])])])])
+    PKG_CHECK_MODULES([CUDA], [cuda-11.1 cudart-11.1],, [
+      PKG_CHECK_MODULES([CUDA], [cuda-11.0 cudart-11.0],, [
+        PKG_CHECK_MODULES([CUDA], [cuda-10.1 cudart-10.1],, [
+          PKG_CHECK_MODULES([CUDA], [cuda-10.0 cudart-10.0],, [
+            PKG_CHECK_MODULES([CUDA], [cuda-9.2 cudart-9.2],, [
+              PKG_CHECK_MODULES([CUDA], [cuda-9.1 cudart-9.1],, [
+                PKG_CHECK_MODULES([CUDA], [cuda-9.0 cudart-9.0],, [
+                  PKG_CHECK_MODULES([CUDA], [cuda-8.0 cudart-8.0],, [
+                    PKG_CHECK_MODULES([CUDA], [cuda-7.5 cudart-7.5],, [
+                      PKG_CHECK_MODULES([CUDA], [cuda-7.0 cudart-7.0],, [
+                        PKG_CHECK_MODULES([CUDA], [cuda-6.5 cudart-6.5],, [
+                          AC_MSG_WARN([Could not find cuda headers/libraries])])])])])])])])])])])])
   fi
 
   HAVE_CUDA_H=no
@@ -977,11 +984,8 @@
 dnl *** NVDEC ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_NVDEC, true)
 AG_GST_CHECK_FEATURE(NVDEC, [nvdec], nvdec, [
-  HAVE_NVCUVID_H=no
   save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CUDA_CFLAGS $save_CPPFLAGS"
-  AC_CHECK_HEADER([nvcuvid.h], [HAVE_NVCUVID_H=yes],
-      AC_MSG_WARN([Could not find nvcuvid.h]))
   CPPFLAGS=$save_CPPFLAGS
 
   HAVE_NVCUVID=no
@@ -991,7 +995,7 @@
       AC_MSG_WARN([Could not find library nvcuvid]))
   LIBS="$save_LIBS"
 
-  if test "x$HAVE_NVCUVID_H" = "xyes" -a "x$HAVE_NVCUVID" = "xyes"; then
+  if test "x$HAVE_NVCUVID" = "xyes"; then
     HAVE_NVDEC=yes
   else
     HAVE_NVDEC=no
diff -Naur gst-plugins-bad-1.14.5.old/sys/nvdec/gstnvdec.c gst-plugins-bad-1.14.5/sys/nvdec/gstnvdec.c
--- gst-plugins-bad-1.14.5.old/sys/nvdec/gstnvdec.c	2020-11-30 13:42:27.635444480 +0100
+++ gst-plugins-bad-1.14.5/sys/nvdec/gstnvdec.c	2020-12-08 17:11:38.332964892 +0100
@@ -31,9 +31,6 @@
 
 #include "gstnvdec.h"
 
-#include <gst/gl/gstglfuncs.h>
-#include <cudaGL.h>
-
 typedef enum
 {
   GST_NVDEC_QUEUE_ITEM_TYPE_SEQUENCE,
diff -Naur gst-plugins-bad-1.14.5.old/sys/nvdec/gstnvdec.h gst-plugins-bad-1.14.5/sys/nvdec/gstnvdec.h
--- gst-plugins-bad-1.14.5.old/sys/nvdec/gstnvdec.h	2020-11-30 13:42:27.635444480 +0100
+++ gst-plugins-bad-1.14.5/sys/nvdec/gstnvdec.h	2020-12-08 17:11:38.332964892 +0100
@@ -29,7 +29,9 @@
 #define __GST_NVDEC_H__
 
 #include <gst/gl/gl.h>
-#include <nvcuvid.h>
+#include <gst/gl/gstglfuncs.h>
+#include "nvcuvid.h"
+#include <cudaGL.h>
 
 G_BEGIN_DECLS
 
diff -Naur gst-plugins-bad-1.14.5.old/sys/nvdec/Makefile.am gst-plugins-bad-1.14.5/sys/nvdec/Makefile.am
--- gst-plugins-bad-1.14.5.old/sys/nvdec/Makefile.am	2020-11-30 13:42:27.635444480 +0100
+++ gst-plugins-bad-1.14.5/sys/nvdec/Makefile.am	2020-12-08 17:11:38.330964862 +0100
@@ -19,7 +19,8 @@
 	$(GST_GL_LIBS) \
 	$(GST_PBUTILS_LIBS) \
 	$(GST_VIDEO_LIBS) \
-	$(GST_LIBS) \
-	$(CUDA_LIBS) -lnvcuvid
+	$(GST_LIBS)
+
+libgstnvdec_la_LIBADD += $(CUDA_LIBS) -lnvcuvid
 
 libgstnvdec_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
diff -Naur gst-plugins-bad-1.14.5.old/sys/nvenc/gstnvbaseenc.c gst-plugins-bad-1.14.5/sys/nvenc/gstnvbaseenc.c
--- gst-plugins-bad-1.14.5.old/sys/nvenc/gstnvbaseenc.c	2020-11-30 13:42:27.627444362 +0100
+++ gst-plugins-bad-1.14.5/sys/nvenc/gstnvbaseenc.c	2020-12-08 17:14:00.233080377 +0100
@@ -1018,6 +1018,12 @@
             nv_ret);
       }
 
+      nv_ret = cuMemFree ((CUdeviceptr) in_gl_resource->cuda_pointer);
+      if (nv_ret != NV_ENC_SUCCESS) {
+        GST_ERROR_OBJECT (nvenc, "Failed to free CUDA device memory, ret %d",
+            nv_ret);
+      }
+
       g_free (in_gl_resource);
       cuCtxPopCurrent (NULL);
     } else
diff -Naur gst-plugins-bad-1.14.5.old/sys/nvenc/gstnvenc.c gst-plugins-bad-1.14.5/sys/nvenc/gstnvenc.c
--- gst-plugins-bad-1.14.5.old/sys/nvenc/gstnvenc.c	2020-11-30 13:42:27.627444362 +0100
+++ gst-plugins-bad-1.14.5/sys/nvenc/gstnvenc.c	2020-12-08 17:01:31.010898633 +0100
@@ -285,9 +285,13 @@
   for (i = 0; i < dev_count; ++i) {
     if (cuDeviceGet (&cdev, i) == CUDA_SUCCESS
         && cuDeviceGetName (name, sizeof (name), cdev) == CUDA_SUCCESS
-        && cuDeviceComputeCapability (&maj, &min, cdev) == CUDA_SUCCESS) {
-      GST_INFO ("GPU #%d supports NVENC: %s (%s) (Compute SM %d.%d)",
-          i, (((maj << 4) + min) >= 0x30) ? "yes" : "no", name, maj, min);
+        && cuDeviceGetAttribute (&maj,
+            CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR, cdev) == CUDA_SUCCESS
+        && cuDeviceGetAttribute (&min,
+            CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
+            cdev) == CUDA_SUCCESS) {
+      GST_INFO ("GPU #%d supports NVENC: %s (%s) (Compute SM %d.%d)", i,
+          (((maj << 4) + min) >= 0x30) ? "yes" : "no", name, maj, min);
       if (i == device_id) {
         cuda_dev = cdev;
       }
